src/controllers/voteController.js

```diff
@@ -0,0 +1,100 @@
+// express async handler
+const asyncHandler = require("express-async-handler");
+
+// vote model
+const Vote = require("../models/voteModel");
+
+// POST /vote/
+const addVote = asyncHandler(async (req, res) => {
+  const { body } = req;
+
+  const add_vote = new Vote(body);
+
+  try {
+    // check if action is valid
+    if (body.action !== "VOTE_UP" && body.action !== "VOTE_DOWN") {
+      throw new Error(
+        "Please provide a valid action, either VOTE_UP or VOTE_DOWN"
+      );
+    }
+
+    // check if user has already voted
+    const checkVote = await Vote.findOne({
+      user_id: body.user_id,
+      content_id: body.content_id,
+    });
+
+    if (checkVote) {
```

Comment:
make it so if they make the request twice it negates the previous voting entry, like if it was `VOTE_UP`, change it to `VOTE_DOWN`, vice versa